modules/Documents/SaveFile.php

```diff
@@ -58,4 +58,36 @@
 			echo 'file_available';
 		}
 }
+
+if(isset($_REQUEST['act']) && $_REQUEST['act'] == 'massDldCnt')
+{
+	$all_files = $_REQUEST['file_id'];
+	$zipfilename = "cache/Documents.zip";
+	$zip = new Vtiger_Zip($zipfilename);
+	if (file_exists($zipfilename)) @unlink($zipfilename);
+	$dec_files =json_decode($all_files,true);
+	foreach ($dec_files as $folder_id => $files_id) {
+		if ($files_id) {
+			$folderQuery = $adb->pquery("SELECT foldername FROM vtiger_attachmentsfolder WHERE folderid = ?", array($folder_id));
+			$folderName = $adb->query_result($folderQuery, 0, 'foldername');
+			$files = explode(";", $files_id);
+			foreach ($files as $file) {
+				if ($file) {
+					$dbQuery = 'SELECT * FROM vtiger_attachments JOIN vtiger_seattachmentsrel ON vtiger_attachments.attachmentsid = vtiger_seattachmentsrel.attachmentsid WHERE crmid = ?' ;
+					$result = $adb->pquery($dbQuery, array($file)) or die('Could not get file list ');
+					if($adb->num_rows($result) == 1) {
+						$pname = @$adb->query_result($result, 0, 'attachmentsid');
+						$name = @$adb->query_result($result, 0, 'name');
+						$filepath = @$adb->query_result($result, 0, 'path');
+						$name = html_entity_decode($name, ENT_QUOTES, $default_charset);
+						$saved_filename = $pname."_".$name;
+						$zip->addFile($filepath.$saved_filename, $folderName.'/'.$saved_filename);
+					}
+				}
+			}
+		}
+	}$zip->save();
+	$zip->forceDownload($zipfilename);
+	unlink($zipfilename);
```

Comment:
I would eliminate this just in case the code deletes the file before the download finishes, you have it controlled on line 67 anyway and it is a cache directory which can be cleaned whenever the system administrator wants.